apps/coordinator/src/components/Wallet/TransactionsTab/FeeBumping/context/FeeBumpContext.tsx

```diff
@@ -0,0 +1,360 @@
+import React, {
+  createContext,
+  useContext,
+  useReducer,
+  ReactNode,
+  useCallback,
+  useState,
+} from "react";
+import { useSelector } from "react-redux";
+
+import {
+  createAcceleratedRbfTransaction,
+  createCancelRbfTransaction,
+  AcceleratedRbfOptions,
+  CancelRbfOptions,
+} from "@caravan/fees";
+
+import { selectWalletConfig } from "selectors/wallet";
+import { MultisigAddressType, Network } from "@caravan/bitcoin";
+
+import {
+  feeBumpingReducer,
+  initialState,
+  FeeBumpingState,
+} from "./feeBumpReducer";
+import {
+  FeeBumpActionTypes,
+  setFeeBumpStatus,
+  setFeeBumpError,
+  setFeeBumpResult,
+} from "./feeBumpActions";
+
+import { FeeBumpStatus, FeeBumpResult } from "../types";
+import {
+  useAnalyzeTransaction,
+  useChangeOutputIndex,
+  useGetAvailableUtxos,
+  useGetGlobalXpubs,
+} from "./hooks";
+
+// =============================================================================
+// CONTEXT TYPE DEFINITION
+// =============================================================================
+
+interface FeeBumpContextType {
+  state: FeeBumpingState;
+  dispatch: React.Dispatch<FeeBumpActionTypes>;
+
+  // Core fee bumping operations (handled internally by context)
+  createAcceleratedRBF: (options?: {
+    changeAddress?: string;
+    feeRate?: number;
+  }) => Promise<FeeBumpResult>;
+  createCancelRBF: (options?: {
+    cancelAddress?: string;
+    feeRate?: number;
+  }) => Promise<FeeBumpResult>;
+
+  // Loading states
+  isCreatingRBF: boolean;
+}
+
+// =============================================================================
+// MAIN CONTEXT PROVIDER
+// =============================================================================
+
+const FeeBumpContext = createContext<FeeBumpContextType | undefined>(undefined);
+
+interface FeeBumpProviderProps {
+  children: ReactNode;
+}
+
+export function FeeBumpProvider({ children }: FeeBumpProviderProps) {
+  // Core state management - never shared with child components
+  const [state, dispatch] = useReducer(feeBumpingReducer, initialState);
+  const [isCreatingRBF, setIsCreatingRBF] = useState(false); // Local loading state for RBF operations
+  // External dependencies
+  const availableUtxos = useGetAvailableUtxos();
+
+  // Getting all needed wallet state
+  const { network, addressType, requiredSigners, totalSigners } =
+    useSelector(selectWalletConfig);
+  const defaultChangeAddress = useSelector(
+    (state: any) => state.wallet?.change?.nextNode?.multisig?.address,
+  );
+
+  const changeOutputIndex = useChangeOutputIndex(state.transaction);
+  const globalXpubs = useGetGlobalXpubs();
+
+  // Use the analysis hook to handle transaction analysis and update state accordingly
+  // this will refresh if underlying data changes (such as transaction, fee estimates, etc.)
+  useAnalyzeTransaction();
```

Comment:
this is the one I have the biggest question mark about. I think this should work as it uses the context and useEffect to analyze if these change and then runs dispatches accordingly. but we'll see. it might cause problems.